clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -815,6 +820,15 @@ public boolean refreshCommittedOffsetsIfNeeded(Timer timer) {
         return null;
     }
 
+    /**
+     * Return the consumer group metadata.
+     *
+     * @return the current consumer group metadata
+     */
+    public ConsumerGroupMetadata groupMetadata() {
+        return new ConsumerGroupMetadata(groupMetadata);
```

Comment:
Why we want to return a clone of the groupMetadata? If we can declare all the fields as final, and whenever we update we always create a new ConsumerGroupMetadata object, then I think we can safely return the cached object directly.

Reply:
Good point, I think we probably don't need to have this.